docs/extended_api/synchronization_primitives/atomic_ref.md

```diff
@@ -0,0 +1,101 @@
+---
+grand_parent: Extended API
+parent: Synchronization Primitives
+nav_order: 1
+---
+
+# `cuda::atomic_ref`
+
+Defined in header `<cuda/atomic>`:
+
+```cuda
+template <typename T, cuda::thread_scope Scope = cuda::thread_scope_system>
+class cuda::atomic_ref;
+```
+
+The class template `cuda::atomic_ref` is an extended form of [`cuda::std::atomic_ref`]
+  that takes an additional [`cuda::thread_scope`] argument, defaulted to
+  `cuda::std::thread_scope_system`.
+It has the same interface and semantics as [`cuda::std::atomic_ref`], with the
+  following additional operations.
+This class additionally deviates from the standard by being backported to C++11.
+
+## Limitations
+
+`cuda::atomic_ref<T>` and `cuda::std::atomic_ref<T>` may only be instantiated with types sized between 4 and 8 bytes.
+
+Constructing multiple `cuda::std::atomic_ref<T, Sco>` with the same reference, but different values for `Sco` is illegal.
+
+`cuda::std::atomic_ref<T>` may be used with structures, but the behavior of performing atomic operations through it on objects with padding bytes is undefined.
```

Comment:
I think this can be combined with the restriction on the object size above:

```suggestion
For `cuda::atomic_ref<T>` and `cuda::std::atomic_ref<T>` the type `T` must satisfy the following:
- `4 <= sizeof(T) <= 8`
- `T` must not have "padding bits", i.e., 'T`'s [object representation](https://en.cppreference.com/w/cpp/language/object#Object_representation_and_value_representation) must not have bits that do not participate in it's value representation  
```